libraries/AP_HAL_ChibiOS/hwdef/common/malloc.c

```diff
@@ -216,45 +216,22 @@ size_t mem_available(void)
     return totalp;
 }
 
-/*
-  realloc implementation thanks to wolfssl, used by AP_Scripting
- */
 void *realloc(void *addr, size_t size)
 {
-    union heap_header *hp;
-    uint32_t prev_size, new_size;
-
-    void *ptr;
-
-    if(addr == NULL) {
-        return chHeapAlloc(NULL, size);
-    }
-
-    /* previous allocated segment is preceded by an heap_header */
-    hp = addr - sizeof(union heap_header);
-    prev_size = hp->used.size; /* size is always multiple of 8 */
-
-    /* check new size memory alignment */
-    if (size % 8 == 0) {
-        new_size = size;
-    } else {
-        new_size = ((int) (size / 8)) * 8 + 8;
+    if (size == 0) {
+       free(addr);
+       return NULL;
     }
-
-    if(prev_size >= new_size) {
-        return addr;
+    if (addr == NULL) {
+        return malloc(size);
     }
-
-    ptr = chHeapAlloc(NULL, size);
-    if (ptr == NULL) {
-        return NULL;
+    void *new_mem= malloc(size);
+    if (new_mem != NULL) {
+        union heap_header *hp = addr - sizeof(union heap_header);
+        memcpy(new_mem, addr, hp->used.size > size ? size : hp->used.size);
     }
-
-    memcpy(ptr, addr, prev_size);
-
-    chHeapFree(addr);
-
-    return ptr;
+    free(addr);
+    return new_mem; union heap_header *hp;
```

Comment:
Is this an error or do I need to learn something?

Reply:
@OXINARF  nice catch

Reply:
*cough* bad rebase/branch swapping. Good catch.